app/views/groups/index.html.erb

```diff
@@ -0,0 +1,32 @@
+<%= render partial: 'shared/navbar/navbar1' %>
+
+<div class="category-page">
+  <div class="category-container">
+  <% @groups.each do |group| %>
+    <div class="category-card">
+      <div class="category-card-details">
+      <%= link_to group_path(group) do %>
+        <% if group.icon.attached? %>
+          <%= image_tag url_for(group.icon), id: "category-icon" %>
+        <% else %>
+          <p>There is no image</p>
+        <% end %>
+      <% end %>
+      
+      <h6><%= group.name %></h6>
+      <p>$<%= number_with_precision(group.expenses.sum(:amount), precision: 2) %></p>
+      </div>
+      <div class="category-card-buttons">
+      <%= link_to "Edit", edit_group_path(group), class:"editbtn" %>
+      <%= button_to "Delete", group_path(group), class:"deletebtn", method: :delete, data: { confirm: "Are you sure?", turbo_confirm: "Are you sure?" } %>
```

Comment:
- [ ] Your delete functionality is not working, please make sure that your functionality works properly if you are giving this option to the user because having such kinds of features which are just for presentation purposes but not working can lead to a `BAD USER EXPERIENCE`.

Reply:
Hi @skyv26,

Thank you for the suggestion.

At the moment the deletion functionality works only if there are no expenses in that category. 

There is a message at the bottom of the page stating that existing expenses in that category needs to be moved or deleted first before removing the category completely.
<img width="261" alt="image" src="https://github.com/George7h/Budget-App/assets/101390796/98119329-733a-4dc9-a52e-da2e5ed6a435">
<img width="275" alt="image" src="https://github.com/George7h/Budget-App/assets/101390796/85560142-0d54-4f29-ae28-9c32fe3260be">
<img width="285" alt="image" src="https://github.com/George7h/Budget-App/assets/101390796/edd43dd6-1554-4098-8c4b-b157b3deb23d">
<img width="305" alt="image" src="https://github.com/George7h/Budget-App/assets/101390796/cb3b28c4-271f-45fd-8031-6272e49ff539">
<img width="275" alt="image" src="https://github.com/George7h/Budget-App/assets/101390796/924c5341-25a9-48dc-a2cb-e09131b4b7b7">
<img width="340" alt="image" src="https://github.com/George7h/Budget-App/assets/101390796/837079c9-1a1b-49ab-a3d0-21dd847fc4da">
<img width="272" alt="image" src="https://github.com/George7h/Budget-App/assets/101390796/6d73be8c-84dc-467e-ade9-1f9d0342e96a">
<img width="277" alt="image" src="https://github.com/George7h/Budget-App/assets/101390796/7123c44c-bab7-433e-8e45-efc6be060f3a">

But I do understand what you are saying. Especially if the user has a lot of categories they might miss the notification and think that it's not working at all.